src/Connection/StreamConnection.php

```diff
@@ -289,7 +289,7 @@ protected function write($buffer)
         $socket = $this->getResource();
```

Comment:
The fix could probably be done here skipping the loop entirely, but I wasn't sure since I could not run tests:
```php
$socket = $this->getResource();
if (!is_resource($socket)) {
    return;
}
```